packages/wonder-blocks-tooltip/src/components/tooltip-popper.tsx

```diff
@@ -25,26 +29,91 @@ type Props = {
     anchorElement?: HTMLElement;
     /** Where should the bubble try to go with respect to its anchor. */
     placement: Placement;
+    /**
+     * Whether the tooltip should automatically update its position when the
+     * anchor element changes.
+     */
+    autoUpdate?: boolean;
 };
 
-export default class TooltipPopper extends React.Component<Props> {
-    _bubbleRefTracker: RefTracker = new RefTracker();
-    _tailRefTracker: RefTracker = new RefTracker();
+/**
+ * A ref tracker for the bubble element.
+ */
+const _bubbleRefTracker: RefTracker = new RefTracker();
+/**
+ * A ref tracker for the tail element.
+ */
+const _tailRefTracker: RefTracker = new RefTracker();
+/**
+ * A MutationObserver that watches for changes to the anchor element.
+ */
+let _observer: MutationObserver | null = null;
+/**
+ * A function that can be called to update the popper.
+ * @see https://popper.js.org/docs/v2/lifecycle/#manual-update
+ */
+let _popperUpdate: PopperUpdateFn | null = null;
```

Comment:
**suggestion:** This feels wrong. Previously when this was a class, these were unique per component instance. Now they are shared by every instance of the component. Is that what we want here?

I think keeping this as a class-based component is the least complex thing - otherwise, i think these have to be `useRef` or `useState` things in the component instance.

Reply:
hmmm yeah that's a good point, I'll revert it back to a class-based component.